File: server/modules/rpc.lua
local nk = require("nakama")

local function get_world_id(_, _)
  local matches = nk.match_list()
  local current_match = matches[1]

  if current_match == nil then
      return nk.match_create("match_handler", { is_world = true })
  else
      return current_match.match_id
  end
end

local game_config = require("game_config")

local function get_game_config(_, _)
  return nk.json_encode(game_config)
end

nk.register_rpc(get_world_id, "get_world_id")

File: server/modules/rpc.go
package main

import (
  "context"
  "database/sql"
  "github.com/heroiclabs/nakama-common/runtime"
)

func rpcGetWorldId(ctx context.Context, logger runtime.Logger, db *sql.DB, nk runtime.NakamaModule, payload string) (string, error) {
  matches, err := nk.MatchList(ctx, 1, true, WorldMatchLabel, getIntPointer(WorldMinPlayers), getIntPointer(WorldMaxPlayers), "");
  if err != nil {
    logger.Error("Match List Error %v", err)
    return "", err
  } else if len(matches) > 0 {
    return matches[0].GetMatchId(), nil
  }

  matchId, err := nk.MatchCreate(ctx, MatchModuleName, nil);
  if err != nil {
    logger.Error("Match Create Error %v", err)
    return "", err
  }

  return matchId, nil
}
